pcl_ros/src/pcl_ros/filters/voxel_grid.cpp

```diff
@@ -57,6 +57,11 @@ pcl_ros::VoxelGrid::filter (const PointCloud2::ConstPtr &input,
                             const IndicesPtr &indices, 
                             PointCloud2 &output)
 {
+  if (!input->width)
+  {
+    return;
```

Comment:
I think we actually want to publish an empty pointcloud in this case so that it still gets passed down the pipeline.

Reply:
Nevermind this is already how it works.

Reply:
In theory this would have different behaviour if the `output` cloud passed by reference is not empty, is that a case we realistically need to address?

Reply:
Not sure, but if we're in doubt, we should keep current or similar behaviour, so we could do:
``` c++
  pcl::PCLPointCloud2 pcl_output;
  pcl_conversions::moveFromPCL(pcl_output, output);
  return;
```